code/game/sound.dm

```diff
@@ -83,6 +83,20 @@ var/list/trayhit_sound = list('sound/items/trayhit1.ogg', 'sound/items/trayhit2.
 
 	var/Dist = world.view + extrarange
 
+	for(var/obj/item/weapon/reagent_containers/R in range(Dist,get_turf(source)))
```

Comment:
why not just do this in the exact opposite way? on reaction, set a var `muted_until` on all turfs in your area; then playsound just returns early if the world time is less than muted_until

Reply:
Thanks - implemented this in the latest commit.